server/src/main/java/org/elasticsearch/index/mapper/SourceFieldMapper.java

```diff
@@ -160,13 +164,21 @@ public static class Builder extends MetadataFieldMapper.Builder {
         private boolean serializeMode;
 
         private final boolean supportsNonDefaultParameterValues;
-
-        public Builder(IndexMode indexMode, final Settings settings, boolean supportsCheckForNonDefaultParams, boolean serializeMode) {
+        private final boolean sourceModeIsNoop;
```

Comment:
Another boolean field... but we need to a way to allow indices created before 9.0.0 to support the _source.mode attribute and only from 9.0.0 it will truly be a no-op.